### DISCUSSION TITLE
Where should hindsight be placed in a repeating loop

### DISCUSSION THREAD

**Discussion by sT0v** [gopro/labs#1190] [Q&A] [answered]
I made this script to record pickleball points/volleys.

`!SAVEpickleBall=$PMSG="PICKLEBALL""$p">p70!S<p70!2E!R`

The decibel trigger and looping works well.

This is great since upon serving the decible trigger condition should be met and stop recording when the volley has ended. What I am missing is where to put the `hindsight` in this code?

The below does not seem to work.
`mV'$p'>p70!S<p70!2EhS5!R`

**Comment by dnewman-gpsw** [comment, marked as answer]
Don't use hindsight 5 seconds in your testing. I'm not sure if that works. Use hS1 for 15s. Set up hindsight out of the loop, as can't be continuously enabled.  ```mVhS1'$p'>p70!S<p70!2E!R5```

**Reply by sT0v** [reply]
Thank you. This seemed to fix my issue. 

`mVhS5>p70!S<p70!2E!R5`

What is the significance of `!R5`?
- with `!R5`- works
- with `!R` - hindsight tries to count up to 5 but can't get past 3s and restarts loop; can't get past trigger

Additionally, how can I best kill a running script without cycling power?

**Reply by dnewman-gpsw** [reply]
!R5 - repeat at character position 5.  It is the go-to command.

**Comment by dnewman-gpsw** [comment]
```mVhS1<r0>p70!S>r0<p70!2E!1R5```

Add recording tests and a one second pause before the repeat to simplify cancelling. If you have *QRDR=1 you can scan !X to cancel

**Reply by sT0v** [reply]
Thank you, I will try that. I had been trying the `$QRDR=1`/`!X` approach but maybe without the pause you suggested it won't register prior to triggering.

So if I chained some scripts, would the `!R5` fail since the char position may be different?

**Reply by dnewman-gpsw** [reply]
Yes, the go-to position is absolute.  If the use Blockly script generator the go-to offsets are automatically computed
